### PR TITLE
Fix a number of issues around margins on last children

### LINKED ISSUE #916: Margins are broken in bullet lists on 0.5.0rc1
### Problem

Margins in bullet lists are extremely large on 0.5.0rc1
Happens on both Sphinx 2.4.4 and Sphinx 3.0.3

### Reproducible Project

RTD 0.4.3, Sphinx 2.4.4 (how it should look):
https://docs.discord.red/en/latest/guide_publish_cogs.html

RTD 0.5.0rc1, Sphinx 2.4.4 & 3.0.3 (broken margins):
https://jack1142-red-prs.readthedocs.io/en/v3-bump_deps/guide_publish_cogs.html

#### Error Logs/Results

RTD 0.5.0rc1, Sphinx 2.4.4 & 3.0.3 (broken margins):
image

#### Expected Results

RTD 0.4.3, Sphinx 2.4.4 (how it should look):
image

### Environment Info

- Python Version: 3.8.0
- Sphinx Version: 2.4.4 & 3.0.3
- RTD Theme Version: 0.5.0rc1

### PR DESCRIPTION
**Pull request by agjohnson**
The HTML4 writer depended on the `.last` selector, a class added by
Sphinx directly on output. Both writers are not using that method
anymore and instead relying on well supported CSS selectors.

This actually also addresses a number of inconsistencies with margin
spacing in lists. Version 0.4.3 and prior have a number of odd issues
around nested lists that are resolved with this change.

Fixes #916

### CODE REVIEW

**Comment by agjohnson** [comment]
A few of the issues resolved here:

Admonitions with inner content that is not a paragraph do not add additional space at the bottom of the admonition. This was due to a bad selector using `:last-child` pseudo selector
image

Inconsistent spacing in bullet lists.

From 0.4.3:
image

From this PR
image

There was additional missing space around elements not matching `<ul><li><p>..` that made the list display too condensed. A proper last-child selector fixes this issue.
https://sphinx-rtd-theme--939.org.readthedocs.build/en/939/demo/lists_tables.html#second-list-level

File: package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "sphinx_rtd_theme",
-  "version": "0.5.0rc1",
+  "version": "0.5.0rc2",
```
Comment (ericholscher) [inline package-lock.json:3]: This seems like it likely shouldn't be here?

Comment (agjohnson) [reply package-lock.json:3]: It should be, it was missed on last release. `setup.cfg` and `package.json` match.

Comment (agjohnson) [reply package-lock.json:3]: Added #940 to catch this next release

**Pull request merged**